cms/tests/test_app_registration.py

```diff
@@ -29,58 +38,73 @@ def tearDown(self):
+        # config autodiscover
         loaded = set([
             str(module) for module in sys.modules
             if 'cms.tests.test_app_registry' in module])
         expected = set([
-            'cms.tests.test_app_registry.app_without_cms_file',
             'cms.tests.test_app_registry',
-            'cms.tests.test_app_registry.app_with_cms_extension.models',
+            'cms.tests.test_app_registry.app_with_cms_feature',
+            'cms.tests.test_app_registry.app_with_cms_feature.apps',
+            'cms.tests.test_app_registry.app_with_cms_feature.cms_apps',
+            'cms.tests.test_app_registry.app_with_cms_feature.models',
+            'cms.tests.test_app_registry.app_without_cms_file',
+            'cms.tests.test_app_registry.app_without_cms_file.apps',
             'cms.tests.test_app_registry.app_without_cms_file.models',
-            'cms.tests.test_app_registry.app_with_cms_extension',
-            'cms.tests.test_app_registry.app_with_cms_extension.cms_apps'
         ])
         self.assertSetEqual(loaded, expected)
 
     @override_settings(INSTALLED_APPS=[
         'cms.tests.test_app_registry.app_with_bad_cms_file',
     ])
     def test_raises_exception_raised_in_cms_file(self):
+        # The cms file intentionally raises a KeyError. We need
+        # to make sure the exception definitely bubbles up and doesn't
+        # get caught.
         with self.assertRaises(KeyError):
             app_registration.autodiscover_cms_configs()
 
     @override_settings(INSTALLED_APPS=[
-        'cms.tests.test_app_registry.app_with_cms_extension',
+        'cms.tests.test_app_registry.app_with_cms_feature',
         'cms.tests.test_app_registry.app_without_cms_file'
     ])
     def test_adds_cms_app_attribute_to_django_app_config(self):
         app_registration.autodiscover_cms_configs()
 
+        # If a cms config is defined, add a cms_app attribute to the
+        # django app config instance. Otherwise don't.
         app_list = [app for app in apps.get_app_configs()]
         self.assertTrue(hasattr(app_list[0], 'cms_app'))
         self.assertEqual(
             app_list[0].cms_app.__class__.__name__, 'CMSSomeFeatureConfig')
         self.assertFalse(hasattr(app_list[1], 'cms_app'))
 
-    @override_settings(INSTALLED_APPS=[
-        'cms.tests.test_app_registry.app_without_cms_app_class',
-    ])
-    def test_raises_exception_when_no_cms_app_class_found_in_cms_file(self):
-        with self.assertRaises(ImproperlyConfigured):
-            app_registration.autodiscover_cms_configs()
-
-    @override_settings(INSTALLED_APPS=[
-        'cms.tests.test_app_registry.app_with_two_cms_app_classes',
-    ])
-    def test_raises_exception_when_more_than_one_cms_app_class_found_in_cms_file(self):
-        with self.assertRaises(ImproperlyConfigured):
-            app_registration.autodiscover_cms_configs()
+    # TODO: Reinstate these tests once decision made with Paulo on
```

Comment:
Personal preference I would label these as a FIXME due to it being broken although it is a TODO too.

Reply:
I'm used to TODO :) Don't mind changing to FIXME if that's the django-cms convention, but will wait to hear what Paulo says and adjust accordingly in the future.